src/NSL/FermionMatrix/Impl/hubbardExp.hpp

```diff
@@ -71,29 +71,29 @@ class HubbardExp : public FermionMatrix<Type,LatticeType> {
         phi_ = phi;
 
         // determine the signs for exp(+/- kappa) exp(+/- i phi)
-        int sgn;
+        int sgn = +1;
         if (this->species == NSL::Hubbard::Species::Particle) {
             sgn = +1;
             this->sgn_kappa_ = +1;
-        } else {
+        } else { // species == NSL::Hubbard::Species::Hole
             sgn = -1;
 
             // On bipartite lattices exp(-kappa) -> exp(kappa)
             // else we need a - sign for holes
-            if(!this->Lat.bipartite()){
-                this->sgn_kappa_ = -1;
-            } else {
+            if(this->Lat.bipartite()){
                 this->sgn_kappa_ = +1;
+            } else {
+                this->sgn_kappa_ = -1;
             }
         }
 
         // calculate exp(+/- i phi)
-        phiExp_ = NSL::LinAlg::exp(
-            sgn*NSL::complex<NSL::RealTypeOf<Type>>{0,1} * phi_ + sgn*mu_
+        this->phiExp_ = NSL::LinAlg::exp(
+            NSL::complex<NSL::RealTypeOf<Type>>{0,sgn} * phi + sgn*mu_
         );
         // calculate exp(+/- phi)^{-1} = exp(-/+ i phi)
-        phiExpInv_ = NSL::LinAlg::exp(
-            sgn*NSL::complex<NSL::RealTypeOf<Type>>{0,-1} * phi_ - sgn*mu_
+        this->phiExpInv_ = NSL::LinAlg::exp(
+            NSL::complex<NSL::RealTypeOf<Type>>{0,-sgn} * phi - sgn*mu_

```

Comment:
Is the purpose of this for M†?  If so then I think we want some complex conjugations in here too.

For example, people study purely-imaginary µ because it doesn't cause a sign problem in the same way.

Reply:
No, this is for `[exp(iφ+µ)]^-1`, rather than `[exp(iφ+µ)]*`, so no conjugations are necessary.